katdal/datasources.py

```diff
@@ -168,37 +167,25 @@ def view_capture_stream(telstate, capture_block_id=None, stream_name=None):
     """
     # Detect the capture block
     if not capture_block_id:
-        capture_blocks = []
-        if 'sdp_capture_block_id' in telstate:
-            for value_time in telstate.get_range('sdp_capture_block_id', st=0):
-                cbid = str(value_time[0])
-                if 'obs_params' in telstate.view(cbid, exclusive=True):
-                    capture_blocks.append(cbid)
-        if not capture_blocks:
+        try:
+            capture_block_id = str(telstate['capture_block_id'])
+        except KeyError:
             raise ValueError('No capture block IDs found in telstate - '
```

Comment:
Off review, but I guess this is now ...block ID found in telstate... (and the same below for ...captured stream found in...) since we only ever have one.

Reply:
Point taken.